### DISCUSSION TITLE
how to get timestamp for each sentence?

### DISCUSSION THREAD

**Discussion by hunter2009pf** [openai/whisper#1348] [Q&A] [answered]
I use whisper to implement autio to text. I get the correct text but without timestamp. How can I modify the codes below so that I can get the timestamp?

        # detect language and transcribe audio
        model = whisper.load_model("base")

        # load audio and pad/trim it to fit 30 seconds
        audio = whisper.load_audio(ASRPage.output_file_path)
        audio = whisper.pad_or_trim(audio)

        # make log-Mel spectrogram and move to the same device as the model
        mel = whisper.log_mel_spectrogram(audio).to(model.device)

        # detect the spoken language
        _, probs = model.detect_language(mel)
        print(f"Detected language: {max(probs, key=probs.get)}")

        # decode the audio
        options = whisper.DecodingOptions()
        result = whisper.decode(model, mel, options)

        # print the recognized text
        print("TEXT: ", result.text)

**Comment by phineas-pta** [comment, marked as answer]
the code from README is a bit over-detailed, if you don't really need low-level controls, you can simply do

```python
import whisper
model = whisper.load_model("base")
audio = whisper.load_audio(ASRPage.output_file_path)
result = model.transcribe(audio)
```

you can then get timestamps using `result["segments"]`

**Reply by hunter2009pf** [reply]
got it, thanks a lot for your help :)

**Reply by berk-efe** [reply]
is there a function to download as .vtt file?

**Reply by matinmrv** [reply]
Hello. what if you try to load a fine-tuned model like this that is trained for your own use case? How can I get the timestamp?
loaded_model = pipeline("automatic-speech-recognition", model=model_path, tokenizer=tokenizer_path, device=0)

(model is saved on my local)
I really appreciate your help.

**Reply by Baka-14** [reply]
Hey did you find a way to do @matinmrv ?
